include/cucumber-cpp/internal/step/StepMacros.hpp

```diff
@@ -25,13 +22,36 @@ ::cucumber::internal::registerStep< step_name >(         \
 )                                                        \
 /**/
 
+#define BODY_WITH_BREAKPOINT_IMPLEMENTATION_(step_class_name)                 \
+  {                                                                           \
+    if (::cucumber::internal::checkAndClearShouldBreakIntoDebugger()) {       \
+      CUKE_BREAK_INTO_DEBUGGER;                                               \
+    }                                                                         \
+    bodyIfCompiledWithBreakpointSupport();                                    \
+  }                                                                           \
+  void step_class_name ::bodyIfCompiledWithBreakpointSupport()
+/**/
+
+#define CUKE_STEP_WITH_BREAKPOINT_WITH_NAME_(step_name, step_matcher)         \
+  CUKE_STEP_WITH_NAME_(step_name, step_matcher)                               \
+  BODY_WITH_BREAKPOINT_IMPLEMENTATION_(step_name)
+/**/
+
+#define CUKE_STEP_(step_matcher)                                              \
+  CUKE_STEP_WITH_BREAKPOINT_WITH_NAME_(CUKE_GEN_OBJECT_NAME_, step_matcher)
+/**/
+
 // ************************************************************************** //
 // **************               GIVEN/WHEN/THEN                ************** //
 // ************************************************************************** //
 
-#define GIVEN CUKE_STEP_
-#define WHEN CUKE_STEP_
-#define THEN CUKE_STEP_
+#define GIVEN CUKE_STEP_WITH_BREAKPOINT_
```

Comment:
Again, did you check that this even compiles? I don't see a definition for this macro, and I don't think you'd need to change it.